connector/controllers/main.py

```diff
@@ -100,7 +100,15 @@ def retry_postpone(job, message, seconds=None):
         except OperationalError as err:
             # Automatically retry the typical transaction serialization errors
             if err.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
+                buff = StringIO()
+                traceback.print_exc(file=buff)
+                _logger.error(buff.getvalue())
+
+                job.set_failed(exc_info=buff.getvalue())
+                with session_hdl.session() as session:
+                    self.job_storage_class(session).store(job)
                 raise
+
```

Comment:
This code is duplicated with the next `except`. Could you add an inner `try: except:` so that the `raise` for OperationalError which are not in PG_CONCURRENCY_ERRORS_TO_RETRY will be trapped by the global `Exception` except? As we'll have one more indentation level, it would be worth to extract the processing.

Something as follows:

```python
try:
    try:
        return self._try_perform_job(session_hdl, job)
    except OperationalError as err:
        if err.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
            raise
        retry_postpone(job, unicode(err), seconds=PG_RETRY)
        [...]
except RetryableJobError as err:
    [...]
except (FailedJobError, Exception):
    [...]
```